### DISCUSSION TITLE
Two body or sub-body element in latex template

### DISCUSSION THREAD

**Discussion by bwbioinfo** [quarto-dev/quarto-cli#3050] [Q&A]
I'm trying to create a simple format that takes inputs in two languages in the body, framed by certain elements that determine language-specific style and spacing. Is there a way to designate sub body elements that can be called from the template? e.g. $body.eng$ and $body.fr$ for language-specific content in English and French.

**Comment by mcanouil** [comment]
If I understand properly, I would use something like what is described in <https://quarto.org/docs/projects/profiles.html#language-content>.

````
---
lang: en
---

::: {lang=en}
{{< include body.eng >}}
:::

::: {lang=fr}
{{< include body.fr >}}
:::
````

**Reply by bwbioinfo** [reply]
Thank you for your reply, I replied below (should have been in reply).

**Comment by bwbioinfo** [comment]
No, I have a template.tex file with language specific headers and I want to define two separate `$body$` s for fr and English, since there are also distinct framing elements that are defined for each. This doesn't give you access to `$body.eng$`/`$body.fr$` in the template. I want both languages on the same document.

![image](https://user-images.githubusercontent.com/5053806/198056084-69640007-459b-46bb-b68a-02cfe4baa5c3.png)

**Reply by mcanouil** [reply]
Then, you probably need to use Lua filters.
I think the best example available to showcase what you need to do is <https://github.com/nmfs-opensci/quarto_titlepages> by @eeholmes.

**Comment by eeholmes** [comment]
@Nihasa Check out https://github.com/nmfs-opensci/quarto_titlepages_v1 This is a version of quarto_titlepages that is for LaTeX users who want to write their own LaTeX template partials.  Your question is a combo of a LaTeX question, pandoc and Quarto.

* How do I get create subsections for different languages? That is a LaTeX question and you'll need to search the LaTeX community help for that. Just google.
* Once you know how to solve this, then you can tackle how to solve with Quarto/Pandoc. There are different solutions depending on what you need to do.
* If you need to get control of the TeX generation, then take a look at the documentation for https://github.com/nmfs-opensci/quarto_titlepages_v1. This will tell you how Pandoc TeX generation works and how to get control of that.

*However...* based just on what you wrote, I think you don't need to get into pandoc templates

* I think you just need to add some packages and babel declarations in your tex header. If this doesn't make sense, do some googling on LaTeX and multi-language support. That will teach you how LaTeX and babel (package) work for multi-language.
* Then in your Quarto doc you'll have code like
<pre>
```{=latex}
whatever code you need to switch languages
```
</pre>
This is basically how language switching works in LaTeX. You switch languages with a command and in Quarto you issue raw latex commands with `{=latex}`.

Also fyi, pandoc often understands LaTeX which means that Quarto understands LaTeX too. So always try first just entering the raw LaTeX into your Quarto doc and see what happens. For example you can create a command with `\newcommand{}` for language switching and then do something like `\makecyrillic{abc}` and it should work.
